crates/common/trie/trie.rs

```diff
@@ -292,7 +292,7 @@ impl Trie {
                         }
                     }
 
-                    node.into()
+                    (*node).into()
```

Comment:
Here we are unboxing the value first, before boxing it again. Maybe we should implement `From<Box<BranchNode>>` for `Node`?